### PR TITLE
Check library crates for all tier 1 targets in PR CI

### PR DESCRIPTION
**Pull request by ChrisDenton**
Let's try checking all tier 1 targets. Shouldn't take much time.

Not sure if this is the right place to put it or not but let's see if it works first.

### CODE REVIEW

**Comment by ChrisDenton** [comment]
Ok, that's working now. And it only takes `1:10` to so seems worth it.

**Comment by RalfJung** [comment]
Cc https://github.com/rust-lang/rust/issues/121519

File: src/ci/docker/host-x86_64/mingw-check/Dockerfile [hunk 1]
```diff
@@ -41,6 +41,8 @@ COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
 ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 
 ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
+           # check library crates on all tier 1 targets
```
Comment (RalfJung) [inline src/ci/docker/host-x86_64/mingw-check/Dockerfile:44]: The list does not include x86_64-linux, I assume because other jobs are handling that? Still, should be explained in a comment.

Same for this compiler-builtins flag, why is that needed?

**Comment by ChrisDenton** [comment]
> Cc https://github.com/rust-lang/rust/issues/121519

Funnily enough I'd somehow completely forgotten about that issue when I first made this PR although it must have been in the back of my mind.

File: src/ci/docker/host-x86_64/mingw-check/Dockerfile (hunk 1 above)
Comment (ChrisDenton) [reply src/ci/docker/host-x86_64/mingw-check/Dockerfile:44]: I've added some comments.

File: src/ci/docker/host-x86_64/mingw-check/Dockerfile [hunk 2]
```diff
@@ -41,6 +41,10 @@ COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
...
+           # We disable optimizing compiler built-ins because that requires building for the target
+           # We also skip the x86_64-unknown-linux-* targets as they're well tested by other jobs.
```
Comment (RalfJung) [inline src/ci/docker/host-x86_64/mingw-check/Dockerfile:46]:
```suggestion
           # We also skip the x86_64-unknown-linux-gnu target as it is well tested by other jobs.
```
There's only one such target, is there?

File: src/ci/docker/host-x86_64/mingw-check/Dockerfile [hunk 3]
```diff
@@ -41,6 +41,10 @@ COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
 ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 
 ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
+           # check library crates on all tier 1 targets
+           # We disable optimizing compiler built-ins because that requires building for the target
```
Comment (RalfJung) [inline src/ci/docker/host-x86_64/mingw-check/Dockerfile:45]:
```suggestion
           # We disable optimizing compiler built-ins because that requires building for the target.
```

Comment (RalfJung) [reply src/ci/docker/host-x86_64/mingw-check/Dockerfile]: It looks like my updated suggestion that I edited in above hasn't been picked up in the latest patch.

Comment (ChrisDenton) [reply src/ci/docker/host-x86_64/mingw-check/Dockerfile]: Oh, it didn't show me the edit until I refreshed. Ho hum.

File: src/ci/docker/host-x86_64/mingw-check/Dockerfile [hunk 4]
```diff
@@ -41,6 +41,10 @@ COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
 ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 
 ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
+           # Check library crates on all tier 1 targets.
+           # We disable optimized compiler built-ins because that requires a C toolchain for the target.
+           # We also skip the x86_64-unknown-linux-gnu target as it is well-tested by other jobs.
+           python3 ../x.py check --stage 0 --set build.optimized-compiler-builtins=false core alloc std --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc && \
```
Comment (Mark-Simulacrum) [inline src/ci/docker/host-x86_64/mingw-check/Dockerfile:47]: I guess this isn't including libtest and proc_macro, but those seem less likely to be target-dependent...

Comment (ChrisDenton) [reply src/ci/docker/host-x86_64/mingw-check/Dockerfile:47]: Yeah. I was going by what tends to fail after an r+. Which is admittedly anecdotal and prone to biasing towards where code changes more frequently. But I figured that if it comes up then it's easy enough to add additional crates.
